src/test/scala/com/microsoft/hyperspace/actions/CreateActionTest.scala

```diff
@@ -104,4 +108,57 @@ class CreateActionTest extends SparkFunSuite with SparkInvolvedSuite {
     assert(
       ex.getMessage.contains(s"Another Index with name ${indexConfig.indexName} already exists"))
   }
+
+  test("Verify rootPaths for given LogicalRelations") {
+    val path1 = sampleParquetDataLocation + "table1"
+    val path2 = sampleParquetDataLocation + "table2"
+    FileUtils.delete(new Path(path1))
+    FileUtils.delete(new Path(path2))
+
+    import spark.implicits._
+    SampleData.testData
+      .toDF("Date", "RGUID", "Query", "imprs", "clicks")
+      .limit(2)
+      .write
+      .parquet(path1)
+
+    SampleData.testData
+      .toDF("Date", "RGUID", "Query", "imprs", "clicks")
+      .limit(3)
+      .write
+      .parquet(path2)
+
+    val workDir = FileContext.getFileContext.getWorkingDirectory.toString
+
+    Seq(
+      (spark.read.format("parquet").load(path1), Seq(path1), 2),
+      (spark.read.format("parquet").load(path1, path2), Seq(path1, path2), 5),
+      (spark.read.parquet(path1), Seq(path1), 2),
+      (spark.read.parquet(path1, path2), Seq(path1, path2), 5),
+      (spark.read.parquet(path1, path1, path1), Seq(path1, path1, path1), 6),
+      (spark.read.format("parquet").option("path", path1).load(path1), Seq(path1), 2),
+      (spark.read.format("parquet").option("path", path1).load(path2), Seq(path2), 3),
+      (spark.read.option("path", path1).parquet(path1), Seq(path1, path1), 4),
+      (spark.read.option("path", path1).parquet(path2), Seq(path1, path2), 5),
+      (
+        spark.read.format("parquet").option("path", path1).load(path1, path2),
+        Seq(path1, path1, path2),
+        7),
+      (spark.read.option("path", path1).parquet(path1, path2), Seq(path1, path1, path2), 7))
+      .foreach {
+        case (df, expectedRootPaths, expectedCount) =>
+          val relation = CreateActionBaseWrapper.getSourceRelations(df)(0)
+          val rootPaths = relation.rootPaths.map {
+            case path if path.startsWith(workDir) => path.drop(workDir.size + 1) // slash
```

Comment:
Do you need this if you change expectedRootPaths to absolute paths?

Reply:
This may not be needed after https://github.com/microsoft/hyperspace/pull/101#discussion_r464622936.